packages/lit-dev-content/site/articles/article/redux-reactive-controllers.md

```diff
@@ -0,0 +1,731 @@
+
+## The Reactive Controller Interface
+
+The Reactive Controller package has two interfaces: one for the controller, `ReactiveController`, and one for the host that it is hooking into, `ReactiveControllerHost`.
+
+### ReactiveController
+
+The `ReactiveController` interface exposes four methods:
+
+- [`hostConnected()`](/docs/api/controllers/#ReactiveController.hostConnected)
+- [`hostDisconnected()`](/docs/api/controllers/#ReactiveController.hostDisconnected)
+- [`hostUpdate()`](/docs/api/controllers/#ReactiveController.hostUpdate)
+- [`hostUpdated()`](/docs/api/controllers/#ReactiveController.hostUpdated)
+
+In Lit `hostConnected()` is called when the host component is placed in the DOM or if the element is already placed in the DOM and the Reactive Controller was just attached. This is a good place to do initialization work when the host component is ready to be used such as adding event listeners.
+
+Similarly `hostDisconnected()` is called when the element is removed from the DOM. This is a good place to do some cleanup work such as removing event listeners.
+
+`hostUpdate()` is called before the element is about to render or re-render. This is a good place to synchronize or compute state before rendering.
+
+`hostUpdated()` is called after an element has just rendered or re-rendered. This is a good place to synchronize or compute a state that is reliant on rendered DOM. It is often discouraged to request an update to the host in this part of the lifecycle unless absolutely necessary as it may cause an unnecessary re-render of the component just after it has already rendered. Request host updates in `hostUpdated()` only when `hostUpdate()` cannot be utilized.
+
+### ReactiveControllerHost
+
+Reactive Controllers typically have access to an instance of an object that implements the `ReactiveControllerHost` interface, which is often passed to them upon initialization. This allows the Reactive Controller to attach itself to the host and request that it update and re-render.
+
+The `ReactiveControllerHost` interface exposes three methods and one property:
+
+- [`addController(controller: ReactiveController)`](/docs/api/ReactiveElement/#ReactiveElement.addController)
+- [`removeController(controller: ReactiveController)`](/docs/api/ReactiveElement/#ReactiveElement.removeController)
+- [`requestUpdate()`](/docs/api/ReactiveElement/#ReactiveElement.requestUpdate)
+- [`updateComplete: Promise<boolean>`](/docs/api/ReactiveElement/#ReactiveElement.updateComplete)
+
+The `addController()` method takes in the controller that you want to hook into the host’s lifecycle.
+
+{% aside "info" "no-header" %}
+
+If the host is already attached to the DOM or rendered onto the page, then `hostConnected()` will be called after attaching the Reactive Controller via `addController()`.
+
+{% endaside %}
```

Comment:
This is specific to `@lit/reactive-element`'s implementation of `addController()`. Perhaps it could be rephrased as a recommended way to implement `addController()` or removed?